CompetitionService

AnatSid opened the pull request:
…g it to the database

AnatSid committed:
Creating a class to create Competition from the JSON format and adding it to the database

KonstantinKizin commented on src/main/java/org/example/CompetitionService.java:
```diff
@@ -0,0 +1,55 @@
+    private final ObjectMapper objectMapper;
+
+    public CompetitionService() {
+        objectMapper = new ObjectMapper();
```
Better to inject ObjectMapper via constructor and not create inside.

KonstantinKizin commented on src/main/java/org/example/CompetitionService.java:
```diff
@@ -0,0 +1,55 @@
+        objectMapper.registerModule(new JavaTimeModule());
+    }
+
+    public Competition createCompetitionFromJson(String jsonRequest) {
```
Not sure that serialisation is responsibility of the service.

AnatSid committed:
Adding new methods to competitionDao

AnatSid committed:
Update method getCompetitonById in DAO class

AnatSid committed:
Create AthleteDAO

KonstantinKizin commented on src/main/java/org/example/athleteService/AthleteDao.java:
```diff
@@ -0,0 +1,168 @@
+    private static final String DATABASE_PASSWORD = "1313";
+
+
+    public void createAthletesInDataBase(Athlete athlete) {
```
no need to mention inDataBase

KonstantinKizin commented on src/main/java/org/example/athleteService/AthleteDao.java:
```diff
@@ -0,0 +1,168 @@
+    }
+
+
+    public List<Athlete> getAthletesByBirthDate(int minDateBirth, int maxDateBirth) {
```
date as int. Looks strange

KonstantinKizin commented on src/main/java/org/example/athleteService/AthleteService.java:
```diff
@@ -0,0 +1,4 @@
+package org.example.athleteService;
+
+public class AthleteService {
```
empty class

KonstantinKizin commented on src/main/java/org/example/athleteService/AthleteService.java:
```diff
@@ -0,0 +1,4 @@
+package org.example.athleteService;
```
why directory has name athleteService ?

KonstantinKizin commented on src/main/java/org/example/competitionService/CompetitionController.java:
```diff
@@ -0,0 +1,23 @@
+
+    public Competition createCompetitionFromJson(String jsonRequest) {
+        try {
+            return objectMapper.readValue(jsonRequest, Competition.class);
```
it is not creation. Actually it is converting json into object. I guess your service should create converted from json Competition

KonstantinKizin commented on src/main/java/org/example/competitionService/CompetitionService.java:
```diff
@@ -0,0 +1,44 @@
+import java.util.List;
+
+
+public class CompetitionService {
```
Where this class is using ?

KonstantinKizin commented on src/main/java/org/example/competitionService/CompetitionService.java:
```diff
@@ -0,0 +1,44 @@
+package org.example.competitionService;
```
why did you created package competitionService ? Just to put one class CompetitionService ?

KonstantinKizin commented on src/main/java/org/example/competitionService/CompetitionService.java:
```diff
@@ -0,0 +1,44 @@
+
+    public void addCompetition(Competition competition) {
+        if(competition == null){
+            throw new RuntimeException("Competition cannot be null");
```
IllegalArgumentException

KonstantinKizin commented on src/main/java/org/example/competitionService/CompetitionService.java:
```diff
@@ -0,0 +1,44 @@
+    }
+
+    public void deleteCompetition(Competition competition) {
+        if (competition.getId() == 0) {
```
what about negative ?

AnatSid committed:
Adding a mapper
Dividing the structure into 3 layers

KonstantinKizin commented on src/main/java/org/example/entity/Race.java:
```diff
@@ -0,0 +1,80 @@
+                '}';
+    }
+
+    public static class CompetitionService {
```
Why the service it is inner class of Race ?

KonstantinKizin commented on src/main/java/org/example/entity/Race.java:
```diff
@@ -0,0 +1,80 @@
+        private static final String DATABASE_URL = "jdbc:postgresql://localhost:5432/swim_base_2";
+        private static final String DATABASE_USER = "postgres";
+        private static final String DATABASE_PASSWORD = "1313";
+        private final ObjectMapper objectMapper;
```
Service should not know about json

KonstantinKizin commented on src/main/java/org/example/entity/Race.java:
```diff
@@ -0,0 +1,80 @@
+
+    public static class CompetitionService {
+
+        private static final String DATABASE_URL = "jdbc:postgresql://localhost:5432/swim_base_2";
```
Service should not know about DB

KonstantinKizin commented on src/main/java/org/example/controller/AthleteController.java:
```diff
@@ -0,0 +1,26 @@
+    private final AthleteService athleteService;
+    private final AthleteJsonMapper athleteJsonMapper;
+
+    public AthleteController(AthleteService athleteService, AthleteJsonMapper athleteJsonMapper) {
```
AthleteJsonMapper - It is Ok, but maybe you can inject just ObjectMapper everywhere ?  Create single ObjectMapper, set all modules and then just inject wherever you need mapping from/to json

KonstantinKizin commented on src/main/java/org/example/dao/AthleteDao.java:
```diff
@@ -0,0 +1,168 @@
+    private static final String DATABASE_PASSWORD = "1313";
+
+
+    public void createAthletes(Athlete athlete) {
```
createAthleteS - > createAthlete 
Or just create. This will be called like athletDao.create(..)  - > this will be quite obvious when you see which dao and which method is calling

AnatSid committed:
-Rename methods id DAO classes
-Object mapper reorganization

AnatSid committed:
Replacing date of birth with LocalTime
(also converted to Postgres with integer on Date)

The add/create method now returns the created object with the required ID

KonstantinKizin commented on src/main/java/org/example/dao/CityDao.java:
```diff
@@ -0,0 +1,97 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class CityDao {
```
you can create AbstractDao where will be located basic logic which is common for all Dao's.

KonstantinKizin commented on src/main/java/org/example/dao/AthleteDao.java:
```diff
@@ -0,0 +1,180 @@
+        }
+    }
+
+    public Athlete getById(long athleteId) {
```
findOne which return Optional

AnatSid committed:
Changing the get bye ID method to find bye ID.
Now returns an Optional instead of an object
